PR: https://github.com/odoo/odoo/pull/

From: 85bcd6fdd86123e26a022d07be21b1e8bc16a771
From: Ravi Patel
Date: 2016-03-09 10:15:32

Structural Changes: 1
Total Changes: 63

[REF] account,payment_paypal: Remove the 'paypal_account' field

This field was relevant before the salepocalypse, it is no longer.
Replaced by generic payment providers instead of putting paypal payment methods
in account.

Remove it from the res.company and res.config model
Remove the method `_migrate_paypal_account` as it was used to find paypal
accounts based on the removed field

Closes #11231, task-22286

================================= pseudo patch: =================================

--- a/addons/payment_paypal/models/__init__.py
+++ b/addons/payment_paypal/models/__init__.py
@@ -1,4 +1,3 @@
 # -*- coding: utf-8 -*-
 
 import paypal
-import res_company

--- a/addons/payment_paypal/models/paypal.py
+++ b/addons/payment_paypal/models/paypal.py
@@ -62,25 +62,6 @@ class AcquirerPaypal(osv.Model):
         'paypal_api_enabled': False,
     }
 
-    def _migrate_paypal_account(self, cr, uid, context=None):
-        """ COMPLETE ME """
-        cr.execute('SELECT id, paypal_account FROM res_company')
-        res = cr.fetchall()
-        for (company_id, company_paypal_account) in res:
-            if company_paypal_account:
-                company_paypal_ids = self.search(cr, uid, [('company_id', '=', company_id), ('provider', '=', 'paypal')], limit=1, context=context)
-                if company_paypal_ids:
-                    self.write(cr, uid, company_paypal_ids, {'paypal_email_account': company_paypal_account}, context=context)
-                else:
-                    paypal_view = self.pool['ir.model.data'].get_object(cr, uid, 'payment_paypal', 'paypal_acquirer_button')
-                    self.create(cr, uid, {
-                        'name': 'Paypal',
-                        'provider': 'paypal',
-                        'paypal_email_account': company_paypal_account,
-                        'view_template_id': paypal_view.id,
-                    }, context=context)
-        return True
-
     def paypal_compute_fees(self, cr, uid, id, amount, currency_id, country_id, context=None):
         """ Compute paypal fees.
 

--- a/addons/payment_paypal/models/res_company.py
+++ b/None
@@ -1,43 +0,0 @@
-# -*- coding: utf-8 -*-
-
-from openerp.osv import fields, osv
-
-
-class ResCompany(osv.Model):
-    _inherit = "res.company"
-
-    def _get_paypal_account(self, cr, uid, ids, name, arg, context=None):
-        Acquirer = self.pool['payment.acquirer']
-        company_id = self.pool['res.users'].browse(cr, uid, uid, context=context).company_id.id
-        paypal_ids = Acquirer.search(cr, uid, [
-            ('website_published', '=', True),
-            ('name', 'ilike', 'paypal'),
-            ('company_id', '=', company_id),
-        ], limit=1, context=context)
-        if paypal_ids:
-            paypal = Acquirer.browse(cr, uid, paypal_ids[0], context=context)
-            return dict.fromkeys(ids, paypal.paypal_email_account)
-        return dict.fromkeys(ids, False)
-
-    def _set_paypal_account(self, cr, uid, id, name, value, arg, context=None):
-        Acquirer = self.pool['payment.acquirer']
-        company_id = self.pool['res.users'].browse(cr, uid, uid, context=context).company_id.id
-        paypal_account = self.browse(cr, uid, id, context=context).paypal_account
-        paypal_ids = Acquirer.search(cr, uid, [
-            ('website_published', '=', True),
-            ('paypal_email_account', '=', paypal_account),
-            ('company_id', '=', company_id),
-        ], context=context)
-        if paypal_ids:
-            Acquirer.write(cr, uid, paypal_ids, {'paypal_email_account': value}, context=context)
-        return True
-
-    _columns = {
-        'paypal_account': fields.function(
-            _get_paypal_account,
-            fnct_inv=_set_paypal_account,
-            nodrop=True,
-            type='char', string='Paypal Account',
-            help="Paypal username (usually email) for receiving online payments."
-        ),
-    }
